Formula/googletest.rb

```diff
@@ -17,6 +18,8 @@ class Googletest < Formula
 
   depends_on "cmake" => :build
 
+  patch :DATA
```

Comment:
Can you upstream this patch?

Reply:
Is this patch even upstream-able (and thus even valid in Homebrew)? `cxx_std_14` is used to define the minimum requirement and probably shouldn't be touched unless upstream wants to increase minimum C++ standard.

From Google documentation, probably not likely until LLVM/Clang 16 is officially released (Mar 7th) which defaults C++17 or 2024 when 10yr mark passes for C++14, Ref: https://opensource.google/documentation/policies/cplusplus-support#support_criteria_4
> _Example: In 2023, if all supported compilers use C++17 as their default, then we can drop support for C++14. Otherwise, we'll need to wait until 2024, which would be 10 years after C++14 was released._

---

If I had to guess, the test would probably pass with existing bottles on Linux as GCC 11 uses C++17 as default.

Can try experimenting with `CMAKE_CXX_STANDARD` or other supported ways of modifying C++ standard.

Reply:
I don't know if it's upstreamable -- I'm more interested in feedback about the consequences of doing this before changing anything about how we build gtest.

Reply:
Would be good to get feedback when used with C++14 projects.

As note, just tested and confirmed that Linux bottle was built with C++17. New test in PR passes with existing bottle.

Reply:
@carlocab Here is an upstream discussion about the issue I encountered: https://github.com/google/googletest/issues/3081

> You have to build gtest with at least C++17 to get the C++17 related matchers/printers.
> You are building the test code with C++20, but the library might have been built with different options.
> You should always build everything with the same compiler options.

I asked in that thread for clarification: https://github.com/google/googletest/issues/3081#issuecomment-1411678958

Reply:
> Can try experimenting with `CMAKE_CXX_STANDARD` or other supported ways of modifying C++ standard, though could have some unwanted consequences depending on how code is written.

Your idea might work. The Nix package manager's googletest uses the approach you suggested (gated behind LLVM 16): https://github.com/NixOS/nixpkgs/pull/207338/commits/a20d967f83cbe369c8026e1f2bd053919cf7aad9